### PR TITLE
Adds signup form validation. Missing parent email

### PR DESCRIPTION
**Pull request by jlong5795**
# Sign Up Form Validation

----Adds field validation to sign up form.----
- Codename checked against RegEx validation
- Birthday verified to be 13+
- Email checked against RegEx validation
- Confirm password field checks against password for character match

- [ ] this checklist is OPTIONAL and serves to show how to format your PR markdown
- [ ] task to implement the feature
- [ ] another task
- [ ] notes on existing tests or plans to test
- [ ] any known problems and plans to solve them

----Any other thoughts...  Need clarification on how to handle sign up for users under 13----

### CODE REVIEW

File: src/components/forms/SignupForm/SignupForm.tsx [hunk 1]
```diff
@@ -42,8 +48,16 @@ export default function SignupForm({
     errorHandler,
   });
 
+  // calculate age
+  function calculate_age(dob: string) {
+    const diff_ms = Date.now() - new Date(dob).getTime();
```
Comment (bramirez96) [inline src/components/forms/SignupForm/SignupForm.tsx]: The rest of our app uses Luxon's `DateTime` object, to be consistent let's use that here as well

File: src/components/forms/SignupForm/SignupForm.tsx [hunk 2]
```diff
@@ -54,15 +68,53 @@ export default function SignupForm({
...
+            },
+          },
+        },
+      })}
+      {authFormInputs.birthday({
+        rules: {
+          validate: {
+            checkAge: (value) => {
+              const age = calculate_age(value);
+              return age > 12 || "Parent's email is required!";
```
Comment (bramirez96) [inline src/components/forms/SignupForm/SignupForm.tsx]: I think i mentioned it in one of the cards and there's an example on the original source code i linked you to, but we actually need a parent email field in this case. There should be a second field that conditionally renders based on the age input by the user

**Review by bramirez96 — CHANGES_REQUESTED** [review]
We can discuss these changes further when we meet, or I can send you a link to the old code that has the proper sort of setup. Mostly looks okay though. I think we also mentioned rendering an error message?

File: src/components/forms/SignupForm/SignupForm.tsx (hunk 2 above)
Comment (jlong5795) [reply src/components/forms/SignupForm/SignupForm.tsx]: 1. The link in the ticket is no longer valid. A merge updated that file and the link is dead. I haven't had a chance to go through commits to find the old version.
2. I wanted to confirm that. The ticket said that was likely the case, but wanted to confirm before I took the time. I can start on that later tonight.

Comment (bramirez96) [reply src/components/forms/SignupForm/SignupForm.tsx]: I'll get you a new link, that's my fault. Look for it on slack in a few minutes

File: src/components/forms/SignupForm/SignupForm.tsx (hunk 1 above)
Comment (jlong5795) [reply src/components/forms/SignupForm/SignupForm.tsx]: Updated to using Luxon

File: src/components/forms/SignupForm/SignupForm.tsx (hunk 2 above)
Comment (jlong5795) [reply src/components/forms/SignupForm/SignupForm.tsx]: Removed and changed to conditionally show parent's email.

**Review by bramirez96 — APPROVED** [review]

**Pull request merged**